### ISSUE TITLE
Why does the image return from dataloader looks strange with repeat

### ISSUE THREAD

**Issue by laurence-lin** [pytorch/pytorch#52544]
I have an image with size (889, 929)

I load the image with dataloader: 

```
train_val_data = {phase: torchvision.datasets.ImageFolder('{}/{}'.format(data_path, phase), data_transforms[phase]) for phase in ['train', 'val']}
dataloaders_dict = {phase: torch.utils.data.DataLoader(train_val_data[phase], 
                             batch_size = BATCH_SIZE, 
                             shuffle = True,
                             num_workers = 2) for phase in ['train', 'val']}
```

The return image have size (889, 929, 3) and I don't know why, my input image is supposed to be grayscale, and when I display single image it proves me that the shape is actually (889, 929)

I tried to show image receive from dataloader by:

for input, label in dataloaders_dict['train']:
  print(input.shape)
  img1 = input.numpy()[0, :, :, :]
  img1 = img1.reshape(889, 929, -1)
  print(img1.shape)

  plt.imshow(img1)
  plt.show()

The return image looks strange as if it repeats:

![image](https://user-images.githubusercontent.com/19323465/108585320-d287f400-7382-11eb-801f-96a3cb357917.png)

My origin image is supposed to look like:

![image](https://user-images.githubusercontent.com/19323465/108585327-dd428900-7382-11eb-871a-d75d72c8e4fb.png)

Why does my image load as grayscale transform to 3 channel and looks strange?

cc @SsnL @VitalyFedyunin @ejguan

**Comment by ptrblck** [comment]
This error is most likely due to a wrong usage of the `reshape` or `view` method, which could interleave the tensor, while a permutation of the dimensions is usually wanted.
I think the forum is a better place to ask such questions.

**Comment by obilaniu** [comment]
This looks shockingly like #52502.

**Comment by laurence-lin** [comment]
I'd found a way to solve the problem, but I'm not sure why it's the cause...

I'm using Colab Notebook, and since I have my GPU on I thought I could set num_workers as 2 for the DataLoader, however this caused the output image to look distorted. 

After I change num_workers to 1 the image is normal now. 

However I can't figure out why this would cause such an effect, if it's not possible to use 2 workers why not just pass an error message of it?

**Comment by SsnL** [comment]
usually this happens when you are not putting the channel dim in correct place

**Comment by VitalyFedyunin** [comment]
What is your `data_transforms` initialization look like? 
Also, you are expected to get 3 channel images from ImageFolder (by default), as we assume that the image folder might have RGB and greyscale images.

**Comment by laurence-lin** [comment]
> What is your `data_transforms` initialization look like?
> Also, you are expected to get 3 channel images from ImageFolder (by default), as we assume that the image folder might have RGB and greyscale images.

Thank you for response! Sorry for late response, the problem is solved now.
As to mention my data_transforms, I simply use transforms.ToTensor() along with transforms.RandomHorizontalFlip().

And I see that for ImageFolder would automatically do convert('RGB') for my grayscale image, I'd finally found out what the problem is. 

Thank you for all you guys help!

**Comment by rahulv0205** [comment]
An alternate solution for the issue is to use permute in the code line "img1 = img1.reshape(889, 929, -1)"  which can be written as "img1 = img1.permute(1,2,0)" probably. It solved the similar issue for me at least.
